From: Julien Muchembled <jm@jmuchemb.eu>
Date: Tue, 29 Apr 2014 11:40:50 +0200
Subject: Add support for TuxOnIce hibernation

systemd does not support non-mainline kernel features so upstream rejected this
patch.
It is however required for systemd integration by tuxonice-userui package.

Forwarded: http://lists.freedesktop.org/archives/systemd-devel/2014-April/018960.html
---
 src/shared/sleep-config.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/shared/sleep-config.c b/src/shared/sleep-config.c
index 2e22bd0..b5050ea 100644
--- a/src/shared/sleep-config.c
+++ b/src/shared/sleep-config.c
@@ -267,6 +267,12 @@ static bool enough_swap_for_hibernation(void) {
         if (getenv_bool("SYSTEMD_BYPASS_HIBERNATION_MEMORY_CHECK") > 0)
                 return true;
 
+        /* TuxOnIce is an alternate implementation for hibernation.
+         * It can be configured to compress the image to a file or an inactive
+         * swap partition, so there's nothing more we can do here. */
+        if (access("/sys/power/tuxonice", F_OK) == 0)
+                return true;
+
         r = find_hibernate_location(NULL, NULL, &size, &used);
         if (r < 0)
                 return false;
